### PR TITLE
Legend derived from DrawingContext.

### PR DESCRIPTION
**Pull request by simzer**

### CODE REVIEW

File: src/chart/main/chart.cpp [hunk 1]
```diff
@@ -105,28 +105,24 @@ void Chart::draw(Gfx::ICanvas &canvas) const
 	if (actPlot
 	    && (!events.draw.begin
 	        || events.draw.begin->invoke(
-	            Util::EventDispatcher::Params{}))) {
+	            Util::EventDispatcher::Params{}))) 
+	{
+		Draw::DrawingContext context(canvas, layout, events.draw, *actPlot);
+
 		Draw::drawBackground(
```
Comment (schaumb) [inline src/chart/main/chart.cpp:112]: Isn't the drawBackground and the drawLabel a DrawingContext too?

File: src/chart/rendering/drawingcontext.h [hunk 2]
```diff
@@ -16,24 +17,28 @@ namespace Draw
...
+	    const Gen::Plot &plot) :
 	    plot(plot),
 	    canvas(canvas),
 	    painter(*static_cast<Painter *>(canvas.getPainter())),
 	    options(*plot.getOptions()),
-	    style(style),
+	    style(plot.getStyle()),
 	    events(events),
-	    boundingRect(rect)
+		layout(layout)
```
Comment (schaumb) [inline src/chart/rendering/drawingcontext.h:31]: space/tab

File: src/chart/rendering/drawlegend.h [hunk 3]
```diff
@@ -1,36 +1,27 @@
...
-	    const Events::Draw::Legend &events,
-	    Gfx::ICanvas &canvas,
+	drawLegend(const DrawingContext &context,
 	    Gen::ChannelId channelType,
 	    double weight);
 
 private:
 	Geom::Rect contentRect;
-	const Gen::Plot &plot;
 	const Events::Draw::Legend &events;
```
Comment (schaumb) [inline src/chart/rendering/drawlegend.h:20]: this member hides the `DrawingContext::events` member.

File: src/chart/main/chart.cpp (hunk 1 above)
Comment (simzer) [reply src/chart/main/chart.cpp:112]: Yes, they will be changed in one of the a following PRs

File: src/chart/rendering/drawlegend.h (hunk 3 above)
Comment (simzer) [reply src/chart/rendering/drawlegend.h:20]: fixed.

File: src/chart/rendering/drawingcontext.h (hunk 2 above)
Comment (simzer) [reply src/chart/rendering/drawingcontext.h]: fixed

**Review by schaumb — APPROVED** [review]

**Pull request merged**